### ISSUE TITLE
Astro JSX Bug - Property body of ArrowFunctionExpression expected node to be of a type ["BlockStatement","Expression"] but instead got "ExpressionStatement"

### ISSUE THREAD

**Issue by oscartbeaumont** [withastro/compiler#780]

**Comment by ematipico** [comment]
Here's what the compiler emits:

```js
return $$render`function DemoWrapper() ${
     return "something"}`;
}, '<stdin>');
export default $$stdin;
```

You can view [the playground](https://live-astro-compiler.vercel.app/?editor-state=WzMxLDEzOSw4LDAsMTYzLDE4Miw4MywxMDAsMiwzLDM3LDIwNCw0OSwxNCwxMjgsMzIsMTIsNzAsMjI1LDE3MSwxNDQsMTI3LDIxMCwxOTYsMTksNTYsMTIzLDcsMjMsMjIsMzUsODUsMjUsMTA0LDczLDQxLDExMywzMiwyMjAsMjIxLDI0LDIyMiwyNTIsMjQyLDUzLDE1NiwxOCw4LDQzLDE3NCwyMDIsMTY3LDY5LDk3LDE4Myw4MSwxNDYsOTMsMTQzLDE1Niw3MywxNjcsMjE3LDUzLDIwNywyMzgsNzksMjAxLDE3MCwxNzgsMjQzLDQwLDE0NiwyMDAsMTU4LDIwMCwxODMsMTM1LDIzMSwxNDIsNSwxNzUsMTA0LDI0LDEyNywxOTIsMTA2LDkwLDEwNSw2NSwyNiwxNjQsMjMzLDE5MywyMjksMTgsNzcsMjMyLDMxLDExNSwxNDksMTM5LDM3LDEwMywwLDAsMF0%3D)

**Comment by bluwy** [comment]
IIUC `.astro` files are HTML-first so it makes sense that writing direct JSX in `.astro` would fail, but it seems like the error comes from the `components/index.tsx` specifically. Is it that Astro's JSX is wrongly used to handle `components/index.tsx`? (Should be Solid's instead)

**Comment by natemoo-re** [comment]
Yeah, it looks like this is a problem with `astro:jsx` being used to handle a Solid component. Maybe our framework detection is getting tripped up somehow?

Either way, this is a JSX bug so I'm moving this issue back from the compiler repo.
